test/unit/test_utils.lua

```diff
@@ -77,4 +78,25 @@ function TestUtils:testDiscoveryOverride()
   lu.assertItemsEquals(opts.discovery, opts_fixture.discovery_override)
 end
 
+function TestUtils:testClearRequestHeaders()
+  -- assign
+  local headers = {}
+
+  _G.ngx = {
+    req = {
+      clear_header = function(header)
+        headers[header] = true
```

Comment:
if the name of the function indicates clearing a value, wouldn't it be better to use a value like `nil` here, rather than a value that's truthy?  i dunno if it's harder to write an assertion for `nil` in lua

Reply:
@brainthinks I see your point but we are just making sure `clear_header` is called with 3 different values. That is all this unit test is testing.

Not sure if it is necessary to make it `nil` for a stubbed function.

Reply:
to me it is counter-intuitive to test that a "cleared value" is true or truthy.  I understand that the value is arbitrary given that this is a mock/stub for a test.  up to you if you want to change it.

Reply:
It's really not a big issue to change; we can certainly switch it to nil if it makes more sense. But, just for understanding, we're not testing whether the cleared value is truthy, we're just using Lua's hash table implementation to store occurrences of a header, and using the boolean (in this case, "true") to in order to align with what the assertion is expecting. I had a tough time coming up with that last sentence as an explanation. Did it make sense?

Reply:
@brainthinks also "Also, the value of an undeclared variable is nil, because there is no special undefined value. "

see https://www.luafaq.org/gotchas.html

So I think making it nil or undefined or blank adds complexity where it isn't necessary.

Again, we are just making sure the method is called at least 3 times with 3 specific values.